public/app/features/plugins/sql/datasource/SqlDatasource.ts

```diff
@@ -131,7 +131,7 @@ export abstract class SqlDatasource extends DataSourceWithBackend<SQLQuery, SQLO
     );
 
     const interpolatedQuery: SQLQuery = {
-      refId: 'tempvar',
+      refId: `tempvar-${Math.random()}`,
```

Comment:
@zoltanbedi looks like you changed this back to 'tempvar'.  Seems if this refId isn't random subsequent calls don't return a response (not sure why).  This is what caused the issue you commented about where datasets weren't persisted when switching visualizations.

Reply:
This is strange. I don't have any better idea tbh, only thing is that because of this change the tests fails.

Reply:
fixed - used the code from your branch